holoviews/plotting/mpl/comms.py

```diff
@@ -0,0 +1,87 @@
+import uuid
+import warnings
+
+try:
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        from matplotlib.backends.backend_nbagg import CommSocket, new_figure_manager_given_figure
+except ImportError:
+    CommSocket = object
+from mpl_toolkits.mplot3d import Axes3D
+
+from ..comms import JupyterComm
+
+mpl_msg_handler = """
+target = $('#{comms_target}');
+img = $('<div />').html(data);
+target.children().each(function () {{ $(this).remove() }})
+target.append(img)
+"""
+
+mpld3_msg_handler = """
+target = $('#fig_el{comms_target}');
+target.children().each(function () {{ $(this).remove() }});
+mpld3.draw_figure("fig_el{comms_target}", data);
+"""
+
+class WidgetCommSocket(CommSocket):
+    """
+    CustomCommSocket provides communication between the IPython
+    kernel and a matplotlib canvas element in the notebook.
+    A CustomCommSocket is required to delay communication
+    between the kernel and the canvas element until the widget
+    has been rendered in the notebook.
+    """
```

Comment:
I believe ``CustomCommSocket`` must be an outdated name. This docstring should mention that ``CommSocket`` is a matplotlib concept that you are customizing to support NbAgg.